### PR TITLE
On Rename make scenario names selectable and fix bugs

### PR DESCRIPTION
**Pull request by arottersman**
## Overview

Connects #1682

Before, when renaming a scenario, the cursor began on the lefthand side of the text. Double clicking to select the whole text just clicked the tab and took you out of rename mode. 

I also noticed two additional bugs:
-  you could set a scenario's name to a blank string. 
- if you pressed the return key without making any changes the `This name already exists` modal came up

This PR starts scenario text off all selected, allows double clicking to select text as you'd expect, and fixes the two bugs above.

## Demo
scenariorename

## Testing Instructions
- rename scenarios and makes sure the text editing behaves like a regular input
- don't make any changes and press Enter -- confirm there's no error
- confirm you can create duplicate named scenarios otherwise
- confirm if you clear out the text and press enter the name goes back to whatever it was

### CODE REVIEW

File: src/mmw/js/src/modeling/views.js [hunk 1]
```diff
@@ -399,7 +399,10 @@ var ScenarioTabPanelView = Marionette.ItemView.extend({
                     return model.get('name').toLowerCase() === newName.toLowerCase();
                 });
 
-                if (match) {
+                if (model.get('name') === newName || !newName){
+                    return false;
+                }
```
Comment (caseycesari) [inline src/mmw/js/src/modeling/views.js]: Unneeded line return here.

File: src/mmw/js/src/modeling/views.js [hunk 2]
```diff
@@ -399,7 +399,10 @@ var ScenarioTabPanelView = Marionette.ItemView.extend({
...
+                else if (match) {
                     console.log('This name is already in use.');
```
Comment (caseycesari) [inline src/mmw/js/src/modeling/views.js]: While you're in here, consider removing this line.

File: src/mmw/js/src/modeling/views.js [hunk 3]
```diff
@@ -411,17 +414,25 @@ var ScenarioTabPanelView = Marionette.ItemView.extend({
...
-                } else if (model.get('name') !== newName) {
+                } else {
                     return model.set('name', newName);
                 }
             },
+            setScenarioName = function(name) {
+                if (!updateScenarioName(self.model, name)) {
+                    self.render(); // resets view state
+                }
+            },
```
Comment (caseycesari) [inline src/mmw/js/src/modeling/views.js]: Code convention here is to add a line break between function definitions.

**Comment by rajadain** [comment]
Consider adding a test to ensure you can't set a scenario name to empty string or just spaces.

**Comment by arottersman** [comment]
Thanks, Casey! Fixed the styling and added some tests.

**Comment by arottersman** [comment]
Going to merge this unless someone would like to take the new tests for a spin

**Comment by rajadain** [comment]
I'll check the tests

**Comment by arottersman** [comment]
Thanks for testing!
